server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/CounterAddHandler.java

```diff
@@ -0,0 +1,161 @@
+        controllers.add(service);
+
+        return controllers;
+    }
+
+    @Override
+    public void removeRuntimeServices(OperationContext context, ModelNode operation, ModelNode containerModel,
+            ModelNode cacheModel) throws OperationFailedException {
+
+        String counterName = getCounterName(operation);
+        String containerName = getContainerName(operation);
+
+        context.removeService(CounterServiceName.CONFIGURATION.getServiceName(containerName, counterName));
+    }
+
+    private ServiceController<?> installCounterService(ServiceTarget target, String containerName,
+            String configurationName, CounterConfiguration configuration) {
+        final InjectedValue<EmbeddedCacheManager> container = new InjectedValue<>();
+        final CounterConfigurationDependencies dependencies = new CounterConfigurationDependencies(container);
+        final Service<?> service = new CounterService(configuration, configurationName, dependencies);
+        final ServiceBuilder<?> builder = target.addService(CounterServiceName.CONFIGURATION.getServiceName(containerName, configurationName), service)
+                .addDependency(CacheContainerServiceName.CACHE_CONTAINER.getServiceName(containerName), EmbeddedCacheManager.class, container);
+        return builder.install();
+    }
+
+    @Override
+    protected void populateModel(ModelNode operation, ModelNode model) throws OperationFailedException {
+        this.populate(operation, model);
+    }
+
+    void populate(ModelNode fromModel, ModelNode toModel) throws OperationFailedException {
+        for (AttributeDefinition attr : CounterResource.ATTRIBUTES) {
+            attr.validateAndSet(fromModel, toModel);
+        }
+    }
+
+    private String getCounterName(ModelNode operation) {
+        PathAddress counterAddress = getCounterAddressFromOperation(operation);
+        return counterAddress.getLastElement().getValue();
+    }
+
+    private String getContainerName(ModelNode operation) {
+        PathAddress containerAddress = getCacheContainerAddressFromOperation(operation);
+        return containerAddress.getLastElement().getValue();
+    }
+
+    private PathAddress getCacheContainerAddressFromOperation(ModelNode operation) {
+        PathAddress counterAddress = getCounterAddressFromOperation(operation);
+        return counterAddress.subAddress(0, counterAddress.size() - 2);
+    }
+
+    private String getCounterType(ModelNode operation) {
+        PathAddress counterAddress = getCounterAddressFromOperation(operation);
+        int size = counterAddress.size();
+        PathAddress subAddress = counterAddress.subAddress(size - 1, size);
+        return subAddress.getLastElement().getKey();
+    }
+
+    private PathAddress getCounterAddressFromOperation(ModelNode operation) {
+        return PathAddress.pathAddress(operation.get(OP_ADDR));
+    }
+
+    private Builder getBuilder(OperationContext context, ModelNode counter, String counterType)
```

Comment:
`context` never used and it can be removed. 
`OperationFailedException` never thrown and it can be removed.